File: dnd5-online-game/android/app/src/main/java/com/example/android/net/dto/AuthDtos.java
package com.example.android.net.dto;

public class AuthDtos {

    private AuthDtos() {}

    public static class RegisterRequest {
        public String username;
        public String password;

        public RegisterRequest(String username, String password) {
            this.username = username;
            this.password = password;
        }
    }

    public static class RegisterResponse {
        public String userId;
        public String username;
    }

    public static class LoginRequest {
        public String username;
        public String password;

        public LoginRequest(String username, String password) {
            this.username = username;
            this.password = password;
        }
    }

    public static class LoginResponse {
        public String accessToken;
        public String tokenType;
        public int expiresInSeconds;
        public UserStatsDto stats;
    }

    public static class UserStatsDto {
        public int wins;
        public int defeats;
        public int masterCount;
    }

    public static class ProfileResponse {
        public String userId;
        public String username;
        public String registrationDate;
        public String theme;
        public UserStatsDto stats;
    }

    public static class UpdateThemeRequest {
        public String theme;

        public UpdateThemeRequest(String theme) { this.theme = theme; }
    }
}

File: dnd5-online-game/backend/Identity/Application/DTOs/AuthDtos.cs
namespace Identity.Application.DTOs
{
    public class RegisterRequest
    {
        public string Username { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
    }

    public class RegisterResponse
    {
        public Guid UserId { get; set; }
        public string Username { get; set; } = string.Empty;
    }

    public class LoginRequest
    {
        public string Username { get; set; } = string.Empty;
        public string Password { get; set; } = string.Empty;
    }

    public class LoginResponse
    {
        public string AccessToken { get; set; } = string.Empty;
        public string TokenType { get; set; } = "Bearer";
        public int ExpiresInSeconds { get; set; }
        public UserStatsDto Stats { get; set; } = new();
    }

    public class UserStatsDto
    {
        public int Wins { get; set; }
        public int Defeats { get; set; }
        public int MasterCount { get; set; }
    }

    public class ProfileResponse
    {
        public Guid UserId { get; set; }
        public string Username { get; set; } = string.Empty;
        public DateTime RegistrationDate { get; set; }
        public string Theme { get; set; } = "light";
        public UserStatsDto Stats { get; set; } = new();
    }

    public class UpdateThemeRequest
    {
        public string Theme { get; set; } = string.Empty;
    }
}
